tests/system/utils.py

```diff
@@ -143,3 +143,75 @@ def convert_pandas_dtypes(df: pd.DataFrame, bytes_col: bool):
     df["numeric_col"] = df["numeric_col"].apply(
         lambda value: decimal.Decimal(str(value)) if value else None  # type: ignore
     )
+
+
+def assert_pandas_df_equal_pca_components(actual, expected, **kwargs):
+    """Compare two pandas dataframes representing PCA components. The columns
+    required to be present in the dataframes are:
+        numerical_value: numeric,
+        categorical_value: List[object(category, value)]
+
+    Args:
+        actual: Actual Pandas DataFrame
+
+        expected: Expected Pandas DataFrame
+
+        kwargs: kwargs to use in `pandas.testing.assert_series_equal` per column
+    """
+    # Compare the index, columns and values separately, as the polarity of the
+    # PCA vectors can be arbitrary
+    pd.testing.assert_index_equal(
+        actual.index, expected.index.astype(actual.index.dtype)
```

Comment:
Thank you for adding this. May worth mentioned index types are ignored in function docstring.